### DISCUSSION TITLE
[RFC Call] Agenda - February 15, 2021

### DISCUSSION THREAD

**Discussion by jonathantneal** [withastro/roadmap#112] [Proposal]
# Agenda

- @FredKSchott - reach consensus - https://github.com/withastro/rfcs/pull/106
- @boehs - open discussion - https://github.com/withastro/rfcs/discussions/112#discussioncomment-2182143
- @natemoo-re - open discussion - https://github.com/withastro/rfcs/discussions/114

--- 

_If there is something that you would like to see discussed, please comment below. We will prioritize pending RFC, but if we have time to spare we will open the call up to get attention on and discuss [other threads and feature requests.](https://github.com/withastro/rfcs/discussions)._

**Comment by boehs** [comment]
# Render directive

A render directive that allows using renderers inline. This would replace the markdown component and the `lang="scss"`/`lang="ts"` attributes, it would also consequently add support for inline react, vue, svelte, etc

## Examples

- `render:md`
- `render:scss`
- `render:ts`
- `render:vue`
- `render:react client:only`?

# Extend https://github.com/withastro/astro/pull/2586 to `layout` and `component`

I'm thinking about the possibility of extending this to the layouts directory. Currently, layouts are treated as components and the directory has no special meaning. This could change to

`layouts/Main.astro` also calls `layouts/Main.svg.ts`. The output of `get()` is it's own file in `dist` - and the props and slots are presented in the file via the `Astro` namespace

---

A similar thing could be done for components, but the result won't end up in `dist`. Instead, you can request the component in a certain style using `<Component version:svg>`

**Comment by FredKSchott** [comment]
@natemoo-re @boehs this thread exists to organize the agenda. Can you both move these posts into their own discussions so that we can consolidate feedback in a single place for each?

**Comment by natemoo-re** [comment]
I'd love to discuss the `is:raw` directive!

## `is:raw` directive (per #114 and https://github.com/withastro/compiler/pull/282)

This is technically not a full RFC, but in the interest of transparency I'd love to get it in front of the community. 

For background, we currently have a special, undocumented `data-astro-raw` attribute that gets special handling in the compiler. This is a primitive that allows us to build things like the `Markdown` component, but as part of our push towards v1.0, I'd like to clean-up and formalize any private APIs that we're using. This is just a renaming and "blessing" of the `data-astro-raw` attribute into an official `is:raw` directive.
